src/org/openstreetmap/josm/plugins/mbtiles/CacheExportPanel.java

```diff
@@ -0,0 +1,233 @@
+					false, 
+					false, 
+					tr("Save layer {0} as...", layerName), 
+					new FileFilter() {
+						@Override
+						public boolean accept(File f) {
+							return f.isDirectory() || f.getName().endsWith(".mbtiles");
+						}
+
+						@Override
+						public String getDescription() {
+							return "MbTiles file (*.mbtiles)";
+						}
+					},
+					JFileChooser.FILES_AND_DIRECTORIES, 
+					null);
+			if (fileDialog == null) {
+				return; // user canceled operation
+			}
+			File saveFile = fileDialog.getSelectedFile();
+
+		    Main.worker.execute(new ExportTask(cache, layerName, saveFile));
+		}
+	}
+
+	@Override
+	public void addGui(PreferenceTabbedPane gui) {
+		JPanel panel = new JPanel();
+		panel.setLayout(new GridBagLayout());
+		CacheAccess<String, BufferedImageCacheEntry> cache = TMSLayer.getCache();
+		DefaultTableModel model = getTableModel(cache);
+		panel.add(getTableForCache(cache, model), GBC.eol().fill(GBC.BOTH));
+		getTabPreferenceSetting(gui).addSubTab(this, "Export MbTiles", new JScrollPane(panel));
+	}
+
+	@Override
+	public boolean ok() {
+		// restart not required
+		return false;
+	}
+
+	@Override
+	public boolean isExpert() {
+		return false;
+	}
+
+	@Override
+	public TabPreferenceSetting getTabPreferenceSetting(PreferenceTabbedPane gui) {
+		return gui.getImageryPreference();
+	}
+	
+	
+    private JTable getTableForCache(final CacheAccess<String, BufferedImageCacheEntry> cache, final TableModel tableModel) {
+        final JTable ret = new JTable(tableModel);
+        ButtonColumn buttonColumn = new ButtonColumn(new CacheExportAction(cache, ret), "Export");
+        TableColumn tableColumn = ret.getColumnModel().getColumn(2);
+        tableColumn.setCellRenderer(buttonColumn);
+        tableColumn.setCellEditor(buttonColumn);
+        return ret;
+    }
+
+    private static DefaultTableModel getTableModel(final CacheAccess<String, BufferedImageCacheEntry> cache) {
+        return new CacheExportPanelTableModel(CacheContentsPanel.getCacheStats(cache), new String[]{tr("Cache name"), tr("Object Count"), tr("Export")});
+    }
+    
+    static class ExportTask extends PleaseWaitRunnable {
+    	private String layerName;
+		private CacheAccess<String, BufferedImageCacheEntry> cache;
+		private File saveFile;
+		private boolean cancel = false;
+
+    	public ExportTask(CacheAccess<String, BufferedImageCacheEntry> cache, String layerName, File saveFile) {
+    		super(tr("Exporing {0} cache files", layerName));
+    		this.layerName = layerName;
+    		this.cache = cache;
+    		this.saveFile = saveFile;
+    	}
+
+
+	@Override
+	protected void realRun() throws IOException {			
+		if (saveFile.exists()) {
+			if (!saveFile.delete()) {
+				throw new IOException("Unable to delete file: " + saveFile.getAbsolutePath());
+			}
+		}
+		Connection connection = null;
+		try {
+			connection = MbTilesUtils.obtainSqliteDbConnection(saveFile, false);
+			connection.setAutoCommit(true);
```

Comment:
The export process would probably be faster if you didn't use auto-commit and instead committed once at the end of the for-loop below.